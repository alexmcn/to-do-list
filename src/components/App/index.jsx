import React from 'react';
import Drawer from '@material-ui/core/Drawer';
import { makeStyles } from '@material-ui/core/styles';

import ListHeader from 'components/ListHeader';
import TodoLists from 'components/TodoLists';
import UserProfile from 'components/UserProfile';

const App = () => {
  const drawerWidth = 280;

  const useStyles = makeStyles(() => ({
    root: {
      display: 'flex',
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
    },
    drawerPaper: {
      width: drawerWidth,
    },
    content: {
      flexGrow: 1,
    },
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <nav className={classes.drawer}>
        <Drawer
          classes={{
            paper: classes.drawerPaper,
          }}
          variant="permanent"
          open
        >
          <UserProfile userName="Guglielmo Marconi" />
          <TodoLists />
        </Drawer>
      </nav>
      <main className={classes.content}>
        <ListHeader title="Team To-Do List" subtitle="Monday 1 July" />
      </main>
    </div>
  );
};

export default App;
