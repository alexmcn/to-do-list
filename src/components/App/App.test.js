import React from 'react';
import renderer from 'react-test-renderer';

import App from './index';

describe('App', () => {
  it('renders without crashing', () => {
    const instance = renderer.create(<App />);

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
