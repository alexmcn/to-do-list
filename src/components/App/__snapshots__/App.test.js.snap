// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="makeStyles-root-1"
>
  <nav
    className="makeStyles-drawer-2"
  >
    <div
      className="MuiDrawer-root MuiDrawer-docked"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-drawerPaper-3 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiAvatar-root makeStyles-avatar-44"
          >
            <img
              alt="Guglielmo Marconi"
              className="MuiAvatar-img"
              src="profile-image.png"
            />
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1"
          />
        </div>
        <div
          className="makeStyles-root-181"
        >
          <nav
            aria-label="Main Lists"
            className="MuiList-root MuiList-padding"
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                >
                  Team To-Do List
                </span>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </nav>
  <main
    className="makeStyles-content-4"
  >
    <div
      className="makeStyles-root-216"
    >
      <h2
        className="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit MuiTypography-gutterBottom"
      >
        Team To-Do List
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit MuiTypography-gutterBottom"
      >
        Monday 1 July
      </p>
    </div>
  </main>
</div>
`;
